REFERENCE.md

```diff
@@ -0,0 +1,138 @@
+# Reference
+
+<!-- DO NOT EDIT: This document was generated by Puppet Strings -->
+
+## Table of Contents
+
+### Classes
+
+* [`hyperglass::agent`](#hyperglassagent): installs the hyperglass linux agent
+* [`hyperglass::server`](#hyperglassserver): installs the hyperglass looking glass
+* [`hyperglass::server::config`](#hyperglassserverconfig): writes the hyperglass config files
+* [`hyperglass::server::dependencies`](#hyperglassserverdependencies): private class that installs all the services hyperglass depends on
+* [`hyperglass::server::install`](#hyperglassserverinstall): installs the hyperglass server
+* [`hyperglass::server::service`](#hyperglassserverservice): manages the hyperglass service + unit file
+
+## Classes
+
+### `hyperglass::agent`
+
+installs the hyperglass linux agent
+
+* **See also**
+  * https://github.com/checktheroads/hyperglass-agent
+
+### `hyperglass::server`
+
+installs the hyperglass looking glass
+
+* **See also**
+  * https://hyperglass.io/
+
+#### Parameters
+
+The following parameters are available in the `hyperglass::server` class.
+
+##### `manage_depended_services`
+
+Data type: `Boolean`
+
+if true, installs all other services that hyperglass requires, like redis, yarn, nginx, python
+
+Default value: ``true``
+
+##### `devices`
+
+Data type: `Hash`
+
+hash containing all the devices hyperglass can connect to. Defaults to demo data so the service starts properly.
+
+Default value: `{
+    'routers' => [
+      {
+        'name' => 'atl_router01',
+        'address' => '10.0.0.2',
+        'network' => {
+          'name' => 'secondary',
+          'display_name' => 'That Other Network',
+        },
+        'credential' => {
+          'username' => 'user2',
+          'password' => ' secret2',
+        },
+        'display_name' => 'Atlanta, GA',
+        'port' => 22,
+        'nos' => 'juniper',
+        'vrfs' => [
+          {
+            'name' => 'default',
+            'display_name' => 'Global',
+            'ipv4' => {
+              'source_address' => '192.0.2.2',
+            },
+          },
+        ],
+      },
+    ],
+  }`
+
+##### `commands`
+
+Data type: `Hash`
+
+specific commands that can be used by the devices
+
+Default value: `{}`
+
+##### `data`
+
+Data type: `Hash`
+
+generic hyperglass configuration hash.
+
+Default value: `{}`
+
+### `hyperglass::server::config`
+
+writes the hyperglass config files
+
+#### Parameters
+
+The following parameters are available in the `hyperglass::server::config` class.
+
+##### `devices`
+
+Data type: `Hash`
+
+hash containing all the devices hyperglass can connect to. Defaults to demo data so the service starts properly.
```

Comment:
eh?

Reply:
sadly the service does not start without a router to connect to. that's why I added the demo data, that's bundled in hyperglass to the $devices parameter. Advantages:
* the installation with puppet is now identical to the manual installation, as both contain the demo data
* since we have the data we can check in our acceptance tests if the service starts properly